dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,311 @@
+                                                      @Nullable final AccessLogWriterFactory writerFactory,
+                                                      @Nullable final MeterRegistry meterRegistry)
+            throws SSLException, CertificateException {
+        Objects.requireNonNull(sb, "builder to decorate must not be null");
+        if (connectorFactory != null) {
+            if (!(connectorFactory instanceof ArmeriaServerDecorator)) {
+                throw new ClassCastException("server.connector.type must be an instance of " +
+                                             ArmeriaServerDecorator.class.getName());
+            }
+            ((ArmeriaServerDecorator) connectorFactory).decorate(sb);
+        }
+        if (meterRegistry != null) {
+            sb.meterRegistry(meterRegistry);
+        }
+        if (writerFactory != null && !writerFactory.getWriter()
+                                                   .equals(AccessLogWriter.disabled())) {
+            LOGGER.trace("Setting up Armeria AccessLogWriter");
+            sb.accessLogWriter(writerFactory.getWriter(), true);
+        } else {
+            LOGGER.info("Armeria access logs will not be written");
+            sb.accessLogWriter(AccessLogWriter.disabled(), true);
+        }
+        return sb;
+    }
+
+    @JsonProperty
+    private @Valid ConnectorFactory connector = ArmeriaHttpConnectorFactory.build();
+    @JsonProperty
+    private @Valid @NotNull AccessLogWriterFactory accessLogWriter = new CommonAccessLogWriterFactory();
+    @JsonProperty
+    private boolean jerseyEnabled = true;
+    @JsonProperty
+    private @MinSize(0) Size maxRequestLength = Size.bytes(Flags.defaultMaxRequestLength());
+    @JsonProperty
+    private boolean disableDateHeader;
+    @JsonProperty
+    private boolean disableServerHeader;
+    @JsonProperty
+    private boolean verboseResponses;
+    @JsonProperty
+    @Nullable
+    private String defaultHostname;
+    @JsonIgnore
+    private transient ServerBuilder serverBuilder;
+
+    public ArmeriaServerFactory() {
+    }
+
+    /**
+     * Sets up the Armeria ServerBuilder with values from the Dropwizard Configuration.
+     * Ref <a href="https://line.github.io/armeria/advanced-production-checklist.html">Production Checklist</a>
+     *
+     * @param serverBuilder A non-production ready ServerBuilder
+     * @return A production-ready ServerBuilder
+     */
+    // visible for testing
+    ServerBuilder decorateServerBuilderFromConfig(final ServerBuilder serverBuilder) {
+        Objects.requireNonNull(serverBuilder);
+        serverBuilder.maxNumConnections(getMaxQueuedRequests());
+        serverBuilder.blockingTaskExecutor(Executors.newFixedThreadPool(getMaxThreads()), true);
+        serverBuilder.maxRequestLength(maxRequestLength.toBytes());
+        serverBuilder.idleTimeoutMillis(getIdleThreadTimeout().toMilliseconds());
+        serverBuilder.gracefulShutdownTimeout(
+                Duration.ofSeconds(30L),
```

Comment:
I guess you can use `getShotdownGracePeriod()` for both quiet period and timeout.